### PR TITLE
Use new cirq transformer API for quantum_chess

### PR DESCRIPTION
**Pull request by dstrain115**
- Convert CircuitTransformer to use __call__ to conform to Transformers
  API.
- Convert SycamoreDecomposer to a free function that conforms to
  Transformers API.

### CODE REVIEW

File: recirq/quantum_chess/circuit_transformer.py [hunk 1]
```diff
@@ -41,7 +41,12 @@ class CircuitTransformer:
     def __init__(self):
         pass
 
-    def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
+    def __call__(
+        self,
+        circuit: cirq.AbstractCircuit,
+        *,
+        context: Optional[cirq.TransformerContext] = None,
+    ) -> cirq.Circuit:
```
Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:49]: I think we can get rid of the `CircuitTransformer` interface now and simply decorate the derived classes with `@cirq.transformer` decorator -- this decorator will verify the that the derived classes adhere to `cirq.TRANSFORMER` interface and also add useful features like automated logging.

File: recirq/quantum_chess/circuit_transformer.py [hunk 2]
```diff
@@ -341,7 +346,12 @@ def qubit_mapping(self, circuit: cirq.Circuit) -> Dict[cirq.Qid, cirq.GridQubit]
         self.mapping = mapping
         return mapping
 
-    def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
+    def __call__(
+        self,
+        circuit: "cirq.AbstractCircuit",
```
Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:351]:
```suggestion
        circuit: cirq.AbstractCircuit,
```

Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:351]:
```suggestion
        circuit: cirq.AbstractCircuit,
```

File: recirq/quantum_chess/circuit_transformer.py [hunk 3]
```diff
@@ -341,7 +346,12 @@ def qubit_mapping(self, circuit: cirq.Circuit) -> Dict[cirq.Qid, cirq.GridQubit]
         self.mapping = mapping
         return mapping
 
-    def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
+    def __call__(
```
Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:349]: Decorate the class with `@cirq.transformer` to add support for automated logging

Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:349]: Decorate the class with `@cirq.transformer` to add support for automated logging

File: recirq/quantum_chess/circuit_transformer.py [hunk 4]
```diff
@@ -386,7 +396,12 @@ def __init__(self, device: cirq.Device):
         super().__init__()
         self.device = device
 
-    def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
+    def __call__(
+        self,
+        circuit: "cirq.AbstractCircuit",
```
Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:401]:
```suggestion
        circuit: cirq.AbstractCircuit,
```

Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:401]:
```suggestion
        circuit: cirq.AbstractCircuit,
```
```suggestion
        circuit: "cirq.AbstractCircuit",
```

File: recirq/quantum_chess/circuit_transformer.py [hunk 5]
```diff
@@ -400,48 +415,52 @@ def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
...
+                ).all_operations()
+            if len(op.qubits) == 3:
+                return cg.optimized_for_sycamore(
+                    cirq.Circuit(cirq.TOFFOLI(*op.controls, *qubits))
+                ).all_operations()
+    return op
+
+
+@cirq.transformer
+def decompose_into_sqrt_iswap(
```
Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:450]: This entire transformer can now be replaced by a single function call in Cirq:
```python
cirq.optimize_for_target_gateset(
    circuit, gateset=cirq.SqrtIswapTargetGateset(), context=cirq.TransformerContext(deep=True)
)
```

As an evidence, here is a colab which tests compilation of all the different cases hardcoded in the `transform_op_into_sqrt_iswap` method above:

https://colab.research.google.com/drive/1sjacBgm1oqa2giTQ_QUlu_GxKqJ9GLsR?resourcekey=0-UIKWzHrpTw58dWE-VPnqlw&usp=sharing

Comment (tanujkhattar) [inline recirq/quantum_chess/circuit_transformer.py:450]: This entire transformer can now be replaced by a single function call in Cirq:
```python
cirq.optimize_for_target_gateset(circuit, gateset=cirq.SqrtIswapTargetGateset(), ignore_failures=False)
```

As an evidence, here is a colab which tests compilation of all the different cases hardcoded in the `transform_op_into_sqrt_iswap` method above:

https://colab.research.google.com/drive/1sjacBgm1oqa2giTQ_QUlu_GxKqJ9GLsR?resourcekey=0-UIKWzHrpTw58dWE-VPnqlw&usp=sharing

**Review by tanujkhattar — CHANGES_REQUESTED** [review]
Looking good overall. Main comment is around using Cirq's inbuilt transformers for compiling circuits to sqrt iswap target gateset, instead of the custom-written transformer here.

**Comment by dstrain115** [comment]
Moved this to unitary repo.

**Pull request closed without merging**